src/components/editor/PropertyTemplateOutline.jsx

```diff
@@ -93,115 +96,136 @@ export class PropertyTemplateOutline extends Component {
+        jsx = <InputLiteral id={key}
                               propertyTemplate={property}
                               key={key}
                               reduxPath={rtReduxPath}
                               rtId={property.rtId} />
-        break;
-
+        break
       case "lookup":
         lookupConfigItem = getLookupConfigItems(property)
-        input = <InputLookupQA propertyTemplate={property}
+        jsx = <InputLookupQA propertyTemplate={property}
                                lookupConfig={lookupConfigItem}
                                rtId = {property.rtId} />
-        break;
-
+        break
       case "resource":
-        if (valueTemplateRefTest(property)){
-          input = []
-          property.valueConstraint.valueTemplateRefs.map((rtId) => {
-            let resourceTemplate = _.find(this.state.nestedResourceTemplates, ['id', rtId])
-            input.push(<div className="row" key={shortid.generate()}>
-              <section className="col-sm-8">
-                <h5>{resourceTemplate.resourceLabel}</h5>
-              </section>
-              <section className="col-sm-4">
-                <PropertyActionButtons handleAddClick={this.handleAddClick}
-                                       handleMintUri={this.handleMintUri} key={shortid.generate()} />
-              </section>
-            </div>)
-            resourceTemplate.propertyTemplates.map((rtProperty) => {
-              let newReduxPath = Object.assign([], rtReduxPath)
-              newReduxPath.push(rtId)
-              newReduxPath.push(rtProperty.propertyURI)
-              const payload = { reduxPath: newReduxPath }
-              if (rtProperty.valueConstraint.defaults && rtProperty.valueConstraint.defaults.length > 0) {
-                payload['defaults'] = []
-                rtProperty.valueConstraint.defaults.map((row, i) => {
-                  payload['defaults'].push({
-                    id: i,
-                    content: row.defaultLiteral,
-                    uri: row.defaultURI
-                  })
-                })
-              }
-              this.props.initNewResourceTemplate(payload)
-              input.push(<PropertyTemplateOutline key={shortid.generate()}
-                                                  propertyTemplate={rtProperty}
-                                                  reduxPath={newReduxPath}
-                                                  initNewResourceTemplate={this.props.initNewResourceTemplate}
-                                                  resourceTemplate={resourceTemplate} />)
-            })
-          })
-          break;
-        }
-        lookupConfigItem = getLookupConfigItem(property)
-        input = <InputListLOC propertyTemplate = {property}
-                              lookupConfig = {lookupConfigItem}
-                              rtId = {property.rtId} />
-
-        break;
+        jsx = this.resourcePropertyJsx(property, rtReduxPath)
```

Comment:
creating a separate method for this code, IMO, makes the code much more readable.  Hopefully the name of the function makes it clear what is happening